sktime/dists_kernels/_base.py

```diff
@@ -0,0 +1,261 @@
+            for arr in X_check:
+                return_X.append(pd.DataFrame(arr))
+        else:
+            arr_check(X_check)
+            return_X = X_check.tolist()
+    elif isinstance(X, list):
+        arr_check(X)
+        return_X = X
+    else:
+        raise TypeError(
+            "X must be a list of pd.DataFrame or numpy of pd.Dataframe or 3d numpy"
+        )
+
+    return return_X
+
+
+class BasePairwiseTransformerPanel(BaseEstimator):
+    """Base pairwise transformer for panel data template class.
+
+    The base pairwise transformer specifies the methods and method
+    signatures that all pairwise transformers have to implement.
+
+    Specific implementations of these methods is deferred to concrete classes.
+    """
+
+    # default tag values - these typically make the "safest" assumption
+    _tags = {
+        "symmetric": False,  # is the transformer symmetric, i.e., t(x,y)=t(y,x) always?
+    }
+
+    def __init__(self):
+        self.symmetric = False
+        super(BasePairwiseTransformerPanel, self).__init__()
+
+    def __call__(self, X, X2=None):
+        """
+        Behaviour: returns pairwise distance/kernel matrix
+            between samples in X and X2 (equal to X if not passed)
+
+        Parameters
+        ----------
+        X: list of pd.DataFrame or 2D np.arrays, of length n
+        X2: list of pd.DataFrame or 2D np.arrays, of length m, optional
+            default X2 = X
+
+        Returns
+        -------
+        distmat: np.array of shape [n, m]
+            (i,j)-th entry contains distance/kernel between X[i] and X2[j]
+
+        Writes to self
+        --------------
+        X_equals_X2: bool = True if X2 was not passed, False if X2 was passed
+            for use to make internal calculations efficient, e.g., in _transform
+        """
+        # no input checks or input logic here, these are done in transform
+        # this just defines __call__ as an alias for transform
+        return self.transform(X=X, X2=X2)
+
+    def transform(self, X, X2=None):
+        """
+        Behaviour: returns pairwise distance/kernel matrix
+            between samples in X and X2 (equal to X if not passed)
+
+        Parameters
+        ----------
+        X: list of pd.DataFrame or 2D np.arrays, of length n
+        X2: list of pd.DataFrame or 2D np.arrays, of length m, optional
+            default X2 = X
+
+        Returns
+        -------
+        distmat: np.array of shape [n, m]
+            (i,j)-th entry contains distance/kernel between X[i] and X2[j]
+
+        Writes to self
+        --------------
+        X_equals_X2: bool = True if X2 was not passed, False if X2 was passed
+            for use to make internal calculations efficient, e.g., in _transform
+        """
+
+        X = _pairwise_panel_x_check(X)
+
+        if X2 is None:
+            X2 = X
+            self.X_equals_X2 = True
+        else:
+            X2 = _pairwise_panel_x_check(X2)
+            self.X_equals_X2 = False
```

Comment:
same as above, i think X can be equal to X2 if X2 is passed.
if you make the change, then update the docstring too